src/plugins/azure/auth.ts

```diff
@@ -0,0 +1,18 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { exec } from "../../util";
+
+export const azLogin = async (subscriptionId: string) => {
+  await exec("az", ["login"], { check: true });
+  await exec("az", ["account", "set", "--subscription", subscriptionId], {
+    check: true,
+  });
```

Comment:
Is setting the subscription necessary? 
`az network bastion` has a `--subscription` option that we could use.
(The performance is a bit slow probably due to the many `az` subprocesses we have to call.)

Reply:
That could be fixed here: https://linear.app/p0-security/issue/ENG-3129/ssh-investigate-how-az-login-works-with-cross-subscription-bastion

We're not 100% sure how cross subscription access will behave.

Reply:
There's another reason why this is necessary—to select the correct Entra ID directory. We do already provide the Bastion's subscription ID to the `az network bastion` command because we provide the whole Bastion resource ID scope string, but in my testing the command would still sometimes fail because `az login` running in non-interactive mode doesn't have the opportunity to ask you which directory to use if you have access to multiple, and if it chooses the wrong one you can get unexpected authorization errors. By calling `az account set --subscription`, we ensure the correct Entra ID directory is in use.

`az account set` reliably takes about 240ms in my testing, so I don't think it's impacting performance much in particular.